Rename stripLeadingAndTrailing[Matched]Characters() to trim()

annevk opened the pull request:
#### f5ea05710cd6cdfe72f6dedc9624f8e11b5e35f1
<pre>
Rename stripLeadingAndTrailing[Matched]Characters() to trim()
<a href="https://bugs.webkit.org/show_bug.cgi?id=257257">https://bugs.webkit.org/show_bug.cgi?id=257257</a>
rdar://109771825

Reviewed by Chris Dumez.

stripWhiteSpace() (although recently removed) set precedent for this
shorter naming and JavaScript has trim() meaning essentially the same
thing.

* Source/JavaScriptCore/runtime/ExceptionHelpers.cpp:
(JSC::invalidParameterInSourceAppender):
(JSC::invalidParameterInstanceofSourceAppender):
* Source/WTF/wtf/Assertions.cpp:
* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::StringImpl::strip):
(WTF::StringImpl::stripLeadingAndTrailingCharacters): Deleted.
* Source/WTF/wtf/text/StringImpl.h:
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::strip const):
(WTF::StringView::stripLeadingAndTrailingMatchedCharacters const): Deleted.
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::String::strip const):
(WTF::String::stripLeadingAndTrailingCharacters const): Deleted.
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp:
(WebCore::ApplicationManifestParser::parseDisplay):
(WebCore::ApplicationManifestParser::parseOrientation):
(WebCore::ApplicationManifestParser::parseIcons):
(WebCore::ApplicationManifestParser::parseGenericString):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::FetchBodyConsumer::packageFormData):
* Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.cpp:
(WebCore::MediaRecorderProvider::isSupported):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::textUnderElement const):
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeSupportsRule):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::updateSubtree):
* Source/WebCore/dom/ScriptElement.cpp:
(WebCore::ScriptElement::determineScriptType const):
* Source/WebCore/editing/TextManipulationController.cpp:
(WebCore::TextManipulationController::replace):
* Source/WebCore/editing/TypingCommand.cpp:
(WebCore::TypingCommand::markMisspellingsAfterTyping):
* Source/WebCore/html/FeaturePolicy.cpp:
(WebCore::processOriginItem):
(WebCore::updateList):
(WebCore::FeaturePolicy::parse):
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::parseLegacyColorValue):
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::extractMIMETypeFromTypeAttributeForLookup):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::parseAcceptAttribute):
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::mediaSessionTitle const):
* Source/WebCore/html/HTMLOptGroupElement.cpp:
(WebCore::HTMLOptGroupElement::groupLabelText const):
* Source/WebCore/html/ValidatedFormListedElement.cpp:
(WebCore::ValidatedFormListedElement::updateVisibleValidationMessage):
* Source/WebCore/html/parser/CSSPreloadScanner.cpp:
(WebCore::hasValidImportConditions):
* Source/WebCore/html/parser/HTMLMetaCharsetParser.cpp:
(WebCore::HTMLMetaCharsetParser::encodingFromMetaAttributes):
* Source/WebCore/html/parser/HTMLPreloadScanner.cpp:
(WebCore::TokenPreloadScanner::StartTagScanner::processImageAndScriptAttribute):
(WebCore::TokenPreloadScanner::StartTagScanner::processAttribute):
(WebCore::TokenPreloadScanner::StartTagScanner::setURLToLoadAllowingReplacement):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::StyleSheetHandler::observeProperty):
(WebCore::StyleSheetHandler::observeComment):
(WebCore::selectorsFromSource):
* Source/WebCore/layout/integration/LayoutIntegrationCoverage.cpp:
(WebCore::LayoutIntegration::printTextForSubtree):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::extractContentLanguageFromHeader):
* Source/WebCore/loader/HTTPHeaderField.cpp:
(WebCore::HTTPHeaderField::create):
* Source/WebCore/page/WindowFeatures.cpp:
(WebCore::parseDisabledAdaptations):
(WebCore::parseDialogFeaturesMap):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::checkMediaType):
* Source/WebCore/platform/ContentType.cpp:
(WebCore::ContentType::parameter const):
(WebCore::splitParameters):
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::truncatedStringForMenuItem):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::isMediaDiskCacheDisabled):
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::platformMaximumBufferSize const):
* Source/WebCore/platform/gtk/SelectionData.cpp:
(WebCore::SelectionData::setURIList):
* Source/WebCore/platform/ios/DragImageIOS.mm:
(WebCore::createDragImageForLink):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::writeURLForTypes):
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType):
* Source/WebCore/platform/mac/PasteboardWriter.mm:
(WebCore::createPasteboardWriter):
* Source/WebCore/platform/network/CacheValidation.cpp:
(WebCore::collectVaryingRequestHeadersInternal):
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::filenameFromHTTPContentDisposition):
(WebCore::parseXFrameOptionsHeader):
(WebCore::parseClearSiteDataHeader):
(WebCore::isForbiddenHeader):
* Source/WebCore/platform/network/HTTPParsers.h:
(WebCore::stripLeadingAndTrailingHTTPSpaces):
* Source/WebCore/platform/network/MIMEHeader.cpp:
(WebCore::retrieveKeyValuePairs):
(WebCore::MIMEHeader::parseHeader):
(WebCore::MIMEHeader::parseContentTransferEncoding):
* Source/WebCore/platform/network/ParsedContentType.cpp:
(WebCore::ParsedContentType::setContentType):
* Source/WebCore/platform/network/ResourceResponseBase.cpp:
(WebCore::ResourceResponseBase::isAttachment const):
(WebCore::ResourceResponseBase::isAttachmentWithFilename const):
* Source/WebCore/platform/network/curl/CookieUtil.cpp:
(WebCore::CookieUtil::parseCookieAttributes):
(WebCore::CookieUtil::parseCookieHeader):
* Source/WebCore/platform/network/curl/CurlMultipartHandle.cpp:
(WebCore::CurlMultipartHandle::extractBoundary):
* Source/WebCore/platform/network/curl/CurlRequest.cpp:
(WebCore::CurlRequest::didReceiveDebugInfo):
* Source/WebCore/platform/network/curl/ResourceResponseCurl.cpp:
(WebCore::ResourceResponse::appendHTTPHeaderField):
(WebCore::ResourceResponse::setStatusLine):
* Source/WebCore/platform/network/soup/ResourceResponseSoup.cpp:
(WebCore::sanitizeFilename):
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::prepareStatementSlow):
(WebCore::SQLiteDatabase::prepareHeapStatementSlow):
* Source/WebCore/platform/text/PlatformLocale.cpp:
(WebCore::Locale::convertFromLocalizedNumber):
* Source/WebCore/platform/win/ClipboardUtilitiesWin.cpp:
(WebCore::extractMarkupFromCFHTML):
(WebCore::fragmentFromCFHTML):
* Source/WebCore/platform/xr/PlatformXR.h:
(PlatformXR::parseSessionFeatureDescriptor):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::setTextFromOption):
* Source/WebCore/svg/SVGAnimationElement.cpp:
(WebCore::parseKeyTimes):
(WebCore::SVGAnimationElement::attributeContainsJavaScriptURL const):
(WebCore::SVGAnimationElement::attributeChanged):
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::parseOffsetValue):
(WebCore::SVGSMILElement::parseClockValue):
(WebCore::SVGSMILElement::parseCondition):
* Source/WebCore/svg/properties/SVGAnimationAdditiveValueFunctionImpl.cpp:
(WebCore::SVGAnimationColorFunction::calculateDistance const):
* Source/WebCore/svg/properties/SVGPropertyTraits.h:
(WebCore::SVGPropertyTraits&lt;Color&gt;::fromString):
(WebCore::SVGPropertyTraits&lt;Color&gt;::parse):
* Source/WebCore/xml/XPathGrammar.cpp:
* Source/WebCore/xml/XPathGrammar.y:
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenuItemPrivate.h:
(_WebKitOptionMenuItem::_WebKitOptionMenuItem):
* Source/WebKit/UIProcess/gtk/WebPopupMenuProxyGtk.cpp:
(WebKit::WebPopupMenuProxyGtk::createPopupMenu):
* Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.cpp:
(WebKit::WebContextMenuClient::searchWithGoogle):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebContextMenuClientMac.mm:
(WebKit::WebContextMenuClient::searchWithGoogle):
* Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/DisplayListRecorderTests.cpp:
* Tools/TestWebKitAPI/cocoa/TestPDFDocument.mm:
(TestWebKitAPI::TestPDFPage::text const):

Canonical link: <a href="https://commits.webkit.org/264597@main">https://commits.webkit.org/264597@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/725ff76907f980eef544369b492861a166f04cdf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8047 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8332 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8544 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9699 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8150 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8051 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10318 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8243 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11015 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8191 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9277 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7340 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9823 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6578 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7365 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14952 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/6864 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7701 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7489 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/10857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/7614 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7959 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6478 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/8214 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7272 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1876 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1933 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11480 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/8435 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7700 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2025 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
Looks fine. I think `trim()` would have been fine too.

cdumez commented on Tools/TestWebKitAPI/cocoa/TestPDFDocument.mm:
```diff
@@ -105,7 +105,7 @@ inline bool shouldStrip(UChar character)
 {
     if (!m_textWithoutSurroundingWhitespace)
-        m_textWithoutSurroundingWhitespace = String(m_page.get().string).stripLeadingAndTrailingCharacters(shouldStrip);
+        m_textWithoutSurroundingWhitespace = String(m_page.get().string).trim(shouldStrip);
```
Should rename to `shouldTrim`.

cdumez commented on Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:
```diff
@@ -960,20 +960,20 @@ TEST(WTF, StringViewStripLeadingAndTrailingMatchedCharacters)
-    EXPECT_TRUE(stringViewFromLiteral("BBBAAAAAA").stripLeadingAndTrailingMatchedCharacters(isA) == stringViewFromLiteral("BBB"));
-    EXPECT_TRUE(stringViewFromLiteral("CCCAAABBB").stripLeadingAndTrailingMatchedCharacters(isA) == stringViewFromLiteral("CCCAAABBB"));
-    EXPECT_TRUE(stringViewFromLiteral("AAAAAAAAA").stripLeadingAndTrailingMatchedCharacters(isA) == StringView::empty());
+    EXPECT_TRUE(stringViewFromLiteral("AAABBBAAA").trim(isA) == stringViewFromLiteral("BBB"));
```
Should rename the test as well.

cdumez commented on Source/WebCore/platform/network/HTTPParsers.h:
```diff
@@ -131,12 +131,12 @@ inline bool isHTTPSpace(UChar character)
 // Strip leading and trailing whitespace as defined in https://fetch.spec.whatwg.org/#concept-header-value-normalize.
 inline String stripLeadingAndTrailingHTTPSpaces(const String& string)
```
Should rename for consistency ideally and update the comment. Though this could be done in a separate patch.

annevk replied:
I want to remove this (and the HTMLSpaces variant) in follow-up patches.

cdumez commented on Source/WebCore/platform/network/HTTPParsers.h:
```diff
@@ -131,12 +131,12 @@ inline bool isHTTPSpace(UChar character)
+    return string.trim(isHTTPSpace);
 }
 
 inline StringView stripLeadingAndTrailingHTTPSpaces(StringView string)
```
ditto.

cdumez commented on Source/WebCore/loader/HTTPHeaderField.cpp:
```diff
@@ -32,13 +32,13 @@ namespace WebCore {
 {
-    StringView strippedName = StringView(unparsedName).stripLeadingAndTrailingMatchedCharacters(isTabOrSpace<UChar>);
-    StringView strippedValue = StringView(unparsedValue).stripLeadingAndTrailingMatchedCharacters(isTabOrSpace<UChar>);
+    auto strippedName = StringView(unparsedName).trim(isTabOrSpace<UChar>);
```
Ideally we'd update the variable name too.

cdumez commented on Source/WebCore/loader/HTTPHeaderField.cpp:
```diff
@@ -32,13 +32,13 @@ namespace WebCore {
-    StringView strippedName = StringView(unparsedName).stripLeadingAndTrailingMatchedCharacters(isTabOrSpace<UChar>);
-    StringView strippedValue = StringView(unparsedValue).stripLeadingAndTrailingMatchedCharacters(isTabOrSpace<UChar>);
+    auto strippedName = StringView(unparsedName).trim(isTabOrSpace<UChar>);
+    auto strippedValue = StringView(unparsedValue).trim(isTabOrSpace<UChar>);
```
ditto.

cdumez commented on Source/WebCore/html/parser/HTMLPreloadScanner.cpp:
```diff
@@ -331,7 +331,7 @@ class TokenPreloadScanner::StartTagScanner {
     void setURLToLoadAllowingReplacement(StringView value)
     {
-        auto strippedURL = value.stripLeadingAndTrailingMatchedCharacters(isASCIIWhitespace<UChar>);
+        auto strippedURL = value.trim(isASCIIWhitespace<UChar>);
```
Would be nice to update the variable name.

cdumez commented on Source/WebCore/editing/TypingCommand.cpp:
```diff
@@ -460,7 +460,7 @@ void TypingCommand::markMisspellingsAfterTyping(Type commandType)
             String strippedPreviousWord;
             if (range && (commandType == TypingCommand::Type::InsertText || commandType == TypingCommand::Type::InsertLineBreak || commandType == TypingCommand::Type::InsertParagraphSeparator || commandType == TypingCommand::Type::InsertParagraphSeparatorInQuotedContent))
-                strippedPreviousWord = plainText(*range).stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline);
+                strippedPreviousWord = plainText(*range).trim(deprecatedIsSpaceOrNewline);
```
Could be nice to update the variable name.

annevk replied:
It seems I can also make this a StringView to avoid an allocation.

cdumez replied:
You'd need to keep the result of `plainText(*range)` alive for the StringView to stay valid so watch out.

cdumez commented on Source/WebCore/editing/TypingCommand.cpp:
```diff
@@ -458,10 +458,10 @@ void TypingCommand::markMisspellingsAfterTyping(Type commandType)
             if (range && (commandType == TypingCommand::Type::InsertText || commandType == TypingCommand::Type::InsertLineBreak || commandType == TypingCommand::Type::InsertParagraphSeparator || commandType == TypingCommand::Type::InsertParagraphSeparatorInQuotedContent))
-                strippedPreviousWord = plainText(*range).stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline);
-            document().editor().markMisspellingsAfterTypingToWord(p1, endingSelection(), !strippedPreviousWord.isEmpty());
+                trimmedPreviousWord = StringView(plainText(*range)).trim(deprecatedIsSpaceOrNewline);
```
This isn't safe

annevk replied:
Interesting and sorry about that. Will revert that part.

cdumez replied:
To clarify, `plainText(*range)` returns a new temporary String. Then you make a StringView to this temporary, then you call trim() which will trim your StringView and returns a StringView as well. On the next line, the string returned by plainText() is dead and your `trimmedPreviousWord` StringView points to a dead String.

cdumez reviewed (changes requested):
Let's stick to renames in this patch. Too easy to sneak in a bug in such a large patch.

cdumez reviewed (approved):
LGTM

annevk committed:
Rename stripLeadingAndTrailing[Matched]Characters() to trim()
https://bugs.webkit.org/show_bug.cgi?id=257257
rdar://109771825

Reviewed by Chris Dumez.

stripWhiteSpace() (although recently removed) set precedent for this
shorter naming and JavaScript has trim() meaning essentially the same
thing.

* Source/JavaScriptCore/runtime/ExceptionHelpers.cpp:
(JSC::invalidParameterInSourceAppender):
(JSC::invalidParameterInstanceofSourceAppender):
* Source/WTF/wtf/Assertions.cpp:
* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::StringImpl::strip):
(WTF::StringImpl::stripLeadingAndTrailingCharacters): Deleted.
* Source/WTF/wtf/text/StringImpl.h:
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::strip const):
(WTF::StringView::stripLeadingAndTrailingMatchedCharacters const): Deleted.
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::String::strip const):
(WTF::String::stripLeadingAndTrailingCharacters const): Deleted.
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp:
(WebCore::ApplicationManifestParser::parseDisplay):
(WebCore::ApplicationManifestParser::parseOrientation):
(WebCore::ApplicationManifestParser::parseIcons):
(WebCore::ApplicationManifestParser::parseGenericString):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::FetchBodyConsumer::packageFormData):
* Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.cpp:
(WebCore::MediaRecorderProvider::isSupported):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::textUnderElement const):
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeSupportsRule):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::updateSubtree):
* Source/WebCore/dom/ScriptElement.cpp:
(WebCore::ScriptElement::determineScriptType const):
* Source/WebCore/editing/TextManipulationController.cpp:
(WebCore::TextManipulationController::replace):
* Source/WebCore/editing/TypingCommand.cpp:
(WebCore::TypingCommand::markMisspellingsAfterTyping):
* Source/WebCore/html/FeaturePolicy.cpp:
(WebCore::processOriginItem):
(WebCore::updateList):
(WebCore::FeaturePolicy::parse):
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::parseLegacyColorValue):
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::extractMIMETypeFromTypeAttributeForLookup):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::parseAcceptAttribute):
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::mediaSessionTitle const):
* Source/WebCore/html/HTMLOptGroupElement.cpp:
(WebCore::HTMLOptGroupElement::groupLabelText const):
* Source/WebCore/html/ValidatedFormListedElement.cpp:
(WebCore::ValidatedFormListedElement::updateVisibleValidationMessage):
* Source/WebCore/html/parser/CSSPreloadScanner.cpp:
(WebCore::hasValidImportConditions):
* Source/WebCore/html/parser/HTMLMetaCharsetParser.cpp:
(WebCore::HTMLMetaCharsetParser::encodingFromMetaAttributes):
* Source/WebCore/html/parser/HTMLPreloadScanner.cpp:
(WebCore::TokenPreloadScanner::StartTagScanner::processImageAndScriptAttribute):
(WebCore::TokenPreloadScanner::StartTagScanner::processAttribute):
(WebCore::TokenPreloadScanner::StartTagScanner::setURLToLoadAllowingReplacement):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::StyleSheetHandler::observeProperty):
(WebCore::StyleSheetHandler::observeComment):
(WebCore::selectorsFromSource):
* Source/WebCore/layout/integration/LayoutIntegrationCoverage.cpp:
(WebCore::LayoutIntegration::printTextForSubtree):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::extractContentLanguageFromHeader):
* Source/WebCore/loader/HTTPHeaderField.cpp:
(WebCore::HTTPHeaderField::create):
* Source/WebCore/page/WindowFeatures.cpp:
(WebCore::parseDisabledAdaptations):
(WebCore::parseDialogFeaturesMap):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::checkMediaType):
* Source/WebCore/platform/ContentType.cpp:
(WebCore::ContentType::parameter const):
(WebCore::splitParameters):
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::truncatedStringForMenuItem):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::isMediaDiskCacheDisabled):
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::platformMaximumBufferSize const):
* Source/WebCore/platform/gtk/SelectionData.cpp:
(WebCore::SelectionData::setURIList):
* Source/WebCore/platform/ios/DragImageIOS.mm:
(WebCore::createDragImageForLink):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::writeURLForTypes):
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType):
* Source/WebCore/platform/mac/PasteboardWriter.mm:
(WebCore::createPasteboardWriter):
* Source/WebCore/platform/network/CacheValidation.cpp:
(WebCore::collectVaryingRequestHeadersInternal):
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::filenameFromHTTPContentDisposition):
(WebCore::parseXFrameOptionsHeader):
(WebCore::parseClearSiteDataHeader):
(WebCore::isForbiddenHeader):
* Source/WebCore/platform/network/HTTPParsers.h:
(WebCore::stripLeadingAndTrailingHTTPSpaces):
* Source/WebCore/platform/network/MIMEHeader.cpp:
(WebCore::retrieveKeyValuePairs):
(WebCore::MIMEHeader::parseHeader):
(WebCore::MIMEHeader::parseContentTransferEncoding):
* Source/WebCore/platform/network/ParsedContentType.cpp:
(WebCore::ParsedContentType::setContentType):
* Source/WebCore/platform/network/ResourceResponseBase.cpp:
(WebCore::ResourceResponseBase::isAttachment const):
(WebCore::ResourceResponseBase::isAttachmentWithFilename const):
* Source/WebCore/platform/network/curl/CookieUtil.cpp:
(WebCore::CookieUtil::parseCookieAttributes):
(WebCore::CookieUtil::parseCookieHeader):
* Source/WebCore/platform/network/curl/CurlMultipartHandle.cpp:
(WebCore::CurlMultipartHandle::extractBoundary):
* Source/WebCore/platform/network/curl/CurlRequest.cpp:
(WebCore::CurlRequest::didReceiveDebugInfo):
* Source/WebCore/platform/network/curl/ResourceResponseCurl.cpp:
(WebCore::ResourceResponse::appendHTTPHeaderField):
(WebCore::ResourceResponse::setStatusLine):
* Source/WebCore/platform/network/soup/ResourceResponseSoup.cpp:
(WebCore::sanitizeFilename):
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::prepareStatementSlow):
(WebCore::SQLiteDatabase::prepareHeapStatementSlow):
* Source/WebCore/platform/text/PlatformLocale.cpp:
(WebCore::Locale::convertFromLocalizedNumber):
* Source/WebCore/platform/win/ClipboardUtilitiesWin.cpp:
(WebCore::extractMarkupFromCFHTML):
(WebCore::fragmentFromCFHTML):
* Source/WebCore/platform/xr/PlatformXR.h:
(PlatformXR::parseSessionFeatureDescriptor):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::setTextFromOption):
* Source/WebCore/svg/SVGAnimationElement.cpp:
(WebCore::parseKeyTimes):
(WebCore::SVGAnimationElement::attributeContainsJavaScriptURL const):
(WebCore::SVGAnimationElement::attributeChanged):
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::parseOffsetValue):
(WebCore::SVGSMILElement::parseClockValue):
(WebCore::SVGSMILElement::parseCondition):
* Source/WebCore/svg/properties/SVGAnimationAdditiveValueFunctionImpl.cpp:
(WebCore::SVGAnimationColorFunction::calculateDistance const):
* Source/WebCore/svg/properties/SVGPropertyTraits.h:
(WebCore::SVGPropertyTraits<Color>::fromString):
(WebCore::SVGPropertyTraits<Color>::parse):
* Source/WebCore/xml/XPathGrammar.cpp:
* Source/WebCore/xml/XPathGrammar.y:
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenuItemPrivate.h:
(_WebKitOptionMenuItem::_WebKitOptionMenuItem):
* Source/WebKit/UIProcess/gtk/WebPopupMenuProxyGtk.cpp:
(WebKit::WebPopupMenuProxyGtk::createPopupMenu):
* Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.cpp:
(WebKit::WebContextMenuClient::searchWithGoogle):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebContextMenuClientMac.mm:
(WebKit::WebContextMenuClient::searchWithGoogle):
* Tools/TestWebKitAPI/Tests/WTF/StringView.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/DisplayListRecorderTests.cpp:
* Tools/TestWebKitAPI/cocoa/TestPDFDocument.mm:
(TestWebKitAPI::TestPDFPage::text const):

Canonical link: https://commits.webkit.org/264597@main
